teps/0048-task-results-without-results.md

```diff
@@ -426,47 +429,77 @@ will fix the `Pipeline`.
 
 ## Proposal
 
-We propose adding an optional field - `default` - to the `Result` specification in `Task`.
+We propose adding default value at the place where variable replacement
+happens and let consumer decide which value it wants to pick up. For example:
 
 ```yaml
 apiVersion: tekton.dev/v1beta1
 kind: Task
 metadata:
   name: task
 spec:
-  results:
-  - name: merge_status
-    description: whether to rebase or squash
-    type: string
-    default: rebase
-  - name: branches
-    description: branches in the repository
-    type: array
-    default:
-    - main
-    - v1alpha1
-    - v1beta1
-    - v1
-  - name: images
-    type: object
-    properties:
-      node:
-       type: string
-       default: "node:latest" 
-      gcloud:
-       type: string
-       default: "gcloud:latest"
   steps:
   ...
+  results:
+    - name: merge_status
+      description: whether to rebase or squash
+      type: string
+    - name: branches
+      description: branches in the repository
+      type: array
+    - name: images
+      type: object
+      properties:
+        node:
+          type: string
+        gcloud:
+          type: string
+---
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: pipeline-with-defaults
+spec:
+  tasks:
```

Comment:
It might be helpful to make this example more realistic based on this TEP's use cases; it would help show why the Pipeline is the appropriate level for defaults rather than the task